sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/json/IJsonNavigator.java

```diff
@@ -0,0 +1,183 @@
+     */
+    boolean tryGetBufferedStringValue(IJsonNavigatorNode stringNode, ByteBuffer value);
+
+    /**
+     * Gets a string value from a node.
+     *
+     * @param stringNode The node to get the string value from.
+     * @return The string value from the node.
+     */
+    String getStringValue(IJsonNavigatorNode stringNode);
+
+    /**
+     * Gets the numeric value for a node as a signed byte.
+     *
+     * @param numberNode The node you want the number value from.
+     * @return A {@code byte} value that represents the number value in the node.
+     */
+    byte getInt8Value(IJsonNavigatorNode numberNode);
+
+    /**
+     * Gets the numeric value for a node as a 16-bit signed integer.
+     *
+     * @param numberNode The node you want the number value from.
+     * @return A {@code short} value that represents the number value in the node.
+     */
+    short getInt16Value(IJsonNavigatorNode numberNode);
+
+    /**
+     * Gets the numeric value for a node as a 32-bit signed integer.
+     *
+     * @param numberNode The node you want the number value from.
+     * @return An {@code int} value that represents the number value in the node.
+     */
+    int getInt32Value(IJsonNavigatorNode numberNode);
+
+    /**
+     * Gets the numeric value for a node as a single precision number if the number is expressed as a floating point.
+     *
+     * @param numberNode The node you want the number value from.
+     * @return A {@code float} that represents the number value in the node.
+     */
+    float getFloat32Value(IJsonNavigatorNode numberNode);
+
+    /**
+     * Gets the numeric value for a node as double precision number if the number is expressed as a floating point.
+     *
+     * @param numberNode The node you want the number value from.
+     * @return A {@code double} that represents the number value in the node.
+     */
+    double getFloat64Value(IJsonNavigatorNode numberNode);
+
+    /**
+     * Gets the UUID value for a node.
+     *
+     * @param uuidNode The node you want the UUID value from.
+     * @return A UUID read from the node.
+     */
+    UUID getUuidValue(IJsonNavigatorNode uuidNode);
+
+    /**
+     * Gets a binary value for a given node from the input.
+     *
+     * @param binaryNode The node to get the binary value from.
+     * @return The binary value from the node.
+     */
+    ByteBuffer getBinaryValue(IJsonNavigatorNode binaryNode);
+
+    /**
+     * Tries to get the buffered binary value from a node.
+     *
+     * @param binaryNode          The node to get the buffered binary from.
+     * @param bufferedBinaryValue The buffered binary value if possible.
+     * @return {@code true} if the JsonNavigator successfully got the buffered binary value; {@code false} if the JsonNavigator failed to get the buffered binary value.
+     */
+    boolean tryGetBufferedBinaryValue(IJsonNavigatorNode binaryNode, ByteBuffer bufferedBinaryValue);
```

Comment:
Are you trying to mimic an out byte[] buffer here? If so, AtomicReference<ByteBuffer> probably works better?

Reply:
Updated all references of ByteBuffer to AtomicReference<byte[]>